heat/core/types.py

```diff
@@ -607,6 +611,42 @@ def can_cast(from_, to, casting="intuitive"):
                 break
 
 
+def iscomplex(x):
+    """
+    Test element-wise if input is complex.
+
+    Parameters
+    ----------
+    x : DNDarray
+
+    Examples
+    --------
+    >>> ht.iscomplex(ht.array([1+1j, 1]))
+    DNDarray([ True, False], dtype=ht.bool, device=cpu:0, split=None)
+    """
+    sanitation.sanitize_in(x)
+
+    if issubclass(x.dtype, _complexfloating):
+        return x.imag != 0
+    else:
+        return factories.zeros(x.shape, bool, split=x.split, device=x.device, comm=x.comm)
```

Comment:
why not use `torch.is_complex`?

Reply:
I did not find it :mag: :smile:

Reply:
torch.is_complex returns the result for the whole tensor, not element-wise

Reply:
wow. that is a really annoying difference. but that is what it is